traits/tests/test_trait_types.py

```diff
@@ -61,3 +61,35 @@ def test_numpy_validators_loaded_if_numpy_present(self):
             shutil.rmtree(tmpdir)
 
         self.assertEqual(output.strip(), "Success")
+
+    def test_default_value_in_init(self):
+        # Tests for the behaviour of the default_value argument
+        # to TraitType.__init__.
+        trait_type = TraitType(default_value=23)
+        self.assertEqual(
+            trait_type.get_default_value(),
+            (DefaultValue.constant, 23),
+        )
+
+        # An explicit default value of None should work as expected.
+        trait_type = TraitType(default_value=None)
+        self.assertEqual(
+            trait_type.get_default_value(),
+            (DefaultValue.constant, None),
+        )
+
+        # If no default is given, get_default_value() returns a value
+        # of Undefined.
+        trait_type = TraitType()
+        self.assertEqual(
+            trait_type.get_default_value(),
+            (DefaultValue.constant, Undefined),
+        )
+
+        # Similarly, if NoDefaultSpecified is given, get_default_value()
+        # is Undefined.
+        trait_type = TraitType(default_value=NoDefaultSpecified)
+        self.assertEqual(
+            trait_type.get_default_value(),
+            (DefaultValue.constant, Undefined),
+        )
```

Comment:
`trait_type = TraitType(default_value=Undefined)` behaves exactly like this, too many sentinels to mean we don't have a default value it seems.

Reply:
Sure. I think that's outside the scope of this PR, though. The main aim of the PR is to restore the behaviour so that there isn't a behaviour change between 6.1 and 6.2. Considering _new_ changes to behaviour can be done separately. :-)

Reply:
I kind of think this is in-scope because this PR also exposes `NoDefaultSpecified` in `api` and recommends using it. But then I guess if we decided that `Undefined` and `NoDefaultSpecified` should be combined we can do `NoDefaultSpecified = Undefined` something like that later and that would be okay.

Reply:
Will concede that this is out of scope and the namespace can be modified later again.